doc/extensions/best-practices.rst

```diff
@@ -37,17 +37,61 @@ Implement each plugin class in a separate Python module
 This keeps CKAN's plugin loading order simple, see :ref:`ckan.plugins`.
 
 
-.. _extension config setting names best practice:
+.. _avoid name clashes:
 
------------------------------------------------------------------
-Names of config settings should include the name of the extension
------------------------------------------------------------------
+------------------
+Avoid name clashes
+------------------
+Many of the names you pick for your identifiers and files must be unique in
+relation to the names used by core CKAN and other extensions. To avoid
+conflicts you should prefix any public name that your extension introduces with
+the name of your extension. For example:
 
-Names of config settings provided by extensions should include the name
-of the extension, to avoid conflicting with core config settings or with
-config settings from other extensions. For example::
+* The names of *configuration settings* introduced by your extension should
+  have the form ``ckan.my_extension.my_config_setting``.
 
-  ckan.my_extension.show_most_popular_groups = True
+* The names of *templates and template snippets* introduced by your extension
+  should begin with the name of your extension::
+
+      snippets/my_extension_useful_snippet.html
+
+  If you have add a lot of templates you can also put them into a separate
+  folder named after your extension instead.
+
+* The names of *template helper functions* introduced by your extension should
+  begin with the name of your extension. For example:
+
+  .. literalinclude:: /../ckanext/example_theme/v08_custom_helper_function/plugin.py
+     :pyobject: ExampleThemePlugin.get_helpers
+
+* The names of *JavaScript modules* introduced by your extension should begin
+  with the name of your extension. For example
+  ``fanstatic/example_theme_popover.js``:
+
+  .. literalinclude:: /../ckanext/example_theme/v16_initialize_a_javascript_module/fanstatic/example_theme_popover.js
+
+* The names of *API action functions* introduced by your extension should begin
+  with the name of your extension. For example
+  ``my_extension_foobarize_everything``.
+
+* The names of *background job queues* introduced by your extension should
+  begin with the name of your extension. For example
+  ``my_extension:super-special-job-queue``.
+
+In some situations, resources like databases may even be shared between
+multiple CKAN *instances*, which requires an even higher degree of uniqueness
```

Comment:
The choice of the word "database" was motivated by the discussion in this PR during which it became clear that sharing a *Redis* database between instances could be useful. But you're probably right that most people will think of the *PostgreSQL* database first when they read "database". I've improved the wording to make things clearer, let me know if that helps.